### DISCUSSION TITLE
How to perform variant prediction in a zero-shot manner (i.e. without providing deep mutational scanning dataset)

### DISCUSSION THREAD

**Discussion by kklarich** [facebookresearch/esm#555] [Q&A]
Hello, 

Thank you so much for your work to make protein models publicly available. 

I am interested in using the zero shot variant prediction model, however I am confused why I need to provide a deep mutational scanning dataset (which I do not have for my protein family of interest) along with the sequence I would like the make a prediction on. Is it possible to run this model without providing a deep mutational scan? If I am reading this figure from the paper correctly, it seems that I shouldn't need to provide to any additional data to the the pre-trained model in order to make predictions, however the example from [this](https://github.com/facebookresearch/esm/tree/main/examples/variant-prediction#zero-shot-variant-prediction-with-protein-language-models) notebook states that a deep mutational scan must be provided with the sequence of interest. 
![Screenshot 2023-05-12 at 2 54 45 PM](https://github.com/facebookresearch/esm/assets/108477838/caf79c04-35f5-4f91-90a5-3022da8f85f1)

**Comment by naailkhan28** [comment]
I haven't used ESM-1v very much but if you have a structure for your sequence (or a predicted structure) you could use ESM-IF to do zero-shot variant prediction using its log likelihood scores - see [the paper](https://proceedings.mlr.press/v162/hsu22a/hsu22a.pdf)

**Reply by kklarich** [reply]
Hmm I'll take a look, I haven't read that paper yet. Thanks!

**Comment by PraljakReps** [comment]
Hello @kklarich, 

They are referring to the mutated sequence variants x_mut, which can be artificially made without requiring assay measurements that correspond to a deep mutation scan. I.e., ESM zero-shot variant prediction does not require assay measurement for each mutation since the model will only compute the log-likelihood ratios. 

However, if you wanted to compute a spearmann rho correlation (to measure model performance), you will need assay measurements.

**Comment by Jeffuuuu** [comment]
Hello everyone,

I am currently working with the ESM Zero-Shot Variant Prediction Model, with the model using MSA-Transformer. My question is : does this model necessarily need a Deep Mutational Scan to work ? And if not, how can you run the algorithm without putting one into it ? 

Thank you in advance for your response.
Jeffuuuu
